src/SearchExpression.h

```diff
@@ -71,11 +74,14 @@ namespace VDMS {
             _db(db), _tag(tag), _or(p_or) {}
 
         void add(PMGD::PropertyPredicate pp) { _predicates.push_back(pp); }
+        void add_e(PMGD::PropertyPredicate pp) { _e_predicates.push_back(pp); }
         const PMGD::StringID tag() const { return _tag; };
         PMGD::Graph &db() const { return _db; }
         const PMGD::PropertyPredicate &predicate(int i) const { return _predicates.at(i); }
         const size_t num_predicates() const { return _predicates.size(); }
 
+        std::vector<PMGD::PropertyPredicate> get_e_pp() const { return _e_predicates; }
+
```

Comment:
Names should be parallel:
get_node_predicate(int i)
num_node_predicates()
get_edge_predicates()

Also for parallelism, it seems you might want to add:
get_node_predicates()
num_edge_predicates()
get_edge_predicate(int i)
but I guess the usage is such that you don't need these, so perhaps it's a foolish consistency.

Reply:
I like consistency too, but I think it is better to signal that those are not needed by not putting them there.